core/src/main/java/org/gbif/pipelines/core/ws/client/internal/GBIFInternalServiceClient.java

```diff
@@ -0,0 +1,174 @@
+  private static final String DATASET_TITLE_KEY = "title";
+  private static final String PUB_ORG_COUNTRY_KEY = "country";
+  private static final String INSTALLATION_TYPE_KEY = "type";
+  private static final String DATASET_NETWORK_KEY = "key";
+  private final GBIFInternalService service;
+  //caches the last requests obtained
+  private final Cache<String, GBIFInternalResponse> datasetResponseCache =
+    CacheBuilder.newBuilder().maximumSize(10000).build();
+
+  /**
+   * create client from provided configuration
+   */
+  public static GBIFInternalServiceClient from(Config wsConfig) {
+    // create client
+    OkHttpClient client = HttpClientFactory.createClient(wsConfig);
+
+    // create service
+    Retrofit retrofit = new Retrofit.Builder().client(client)
+      .baseUrl(wsConfig.getBasePath())
+      .addConverterFactory(GsonConverterFactory.create())
+      .validateEagerly(true)
+      .build();
+
+    return new GBIFInternalServiceClient(retrofit.create(GBIFInternalService.class));
+  }
+
+  /**
+   * initialize Client with default values
+   */
+  public static GBIFInternalServiceClient client() {
+    return GBIFInternalServiceClient.from(HttpConfigFactory.createConfig(Service.GBIF_INTERNAL));
+  }
+
+  private GBIFInternalServiceClient(GBIFInternalService internalService) { this.service = internalService;}
+
+  /**
+   * requests https://api.gbif.org/v1/dataset/{datasetid}/networks
+   *
+   * @return array of networks for provided datasetID
+   */
+  public JsonArray getNetworkFromDataset(String datasetUUID) {
+    Objects.requireNonNull(datasetUUID);
+    return performCall(service.getNetworkFromDataset(datasetUUID)).getAsJsonArray();
+  }
+
+  /**
+   * requests https://api.gbif.org/v1/organisation/{organisationid}
+   *
+   * @return organisation info for provided organisation id
+   */
+  public JsonObject getOrganizationInfo(String organizationUUID) {
+    Objects.requireNonNull(organizationUUID);
+    return performCall(service.getOrganizationInfo(organizationUUID)).getAsJsonObject();
+  }
+
+  /**
+   * requests http://api.gbif.org/v1/dataset/{datasetid}
+   *
+   * @return dataset info for provided dataset id
+   */
+  public JsonObject getDatasetInfo(String datasetUUID) {
+    Objects.requireNonNull(datasetUUID);
+    return performCall(service.getDatasetInfo(datasetUUID)).getAsJsonObject();
+  }
+
+  /**
+   * requests http://api.gbif.org/v1/installation/{installation_id}
+   *
+   * @return installation info
+   */
+  public JsonObject getInstallationInfo(String installationUUID) {
+    Objects.requireNonNull(installationUUID);
+    return performCall(service.getInstallationInfo(installationUUID)).getAsJsonObject();
+  }
+
+  /**
+   * perform webservice call to aggregate the needed GBIF terms
+   *
+   * @param datasetUUID datasetUUID
+   *
+   * @return aggregated GBIFTerms response for the provided datasetUUID
+   */
+  public GBIFInternalResponse getInternalResponse(String datasetUUID) throws ExecutionException {
```

Comment:
ExecutionException should not be here, that is intended to be used by concurrent code only